src/api/types.rs

```diff
@@ -1,14 +1,71 @@
+extern crate chrono;
 
-
-#[derive(Queryable, Serialize, Deserialize)]
+#[derive(Queryable, Insertable, Serialize, Deserialize)]
+#[table_name="test_names"]
 pub struct TestName {
-    pub id: i32,
     pub name: String,
 }
 
+#[derive(Queryable, Insertable)]
+#[table_name="artifact_names"]
+pub struct ArtifactName {
+	pub name: String,
+}
+
+#[derive(Queryable, Insertable)]
+#[table_name="version"]
```

Comment:
shouldn't this be `versions` or make the other tables singular? Personally I prefer singluar table names (but I don't know the standard).